VivistaServer/Templates/dashboard.liquid

```diff
@@ -298,82 +594,192 @@
 		}
+		ParameterChanged();
+		GeneralParameterChanged();
+		GeneralParameterMemoryChanged();
 
 		mixedChart.update();
 		areaChart.update();
+		barChart.update();
 	}
 
+
 	function GetPercentile95Data() {
+		if (selectedData === null) {
+			return null;
+		}
 		return selectedData
 			.map(a => ({ x: a.timestamp, y: a.percentile95 * 1000, endpoint: a.endpoint }));
 	}
 
 	function GetAverageData() {
+		if (selectedData === null) {
+			return null;
+		}
 		return selectedData
 			.map(a => ({ x: a.timestamp, y: a.average * 1000, endpoint: a.endpoint }));
 	}
 
 	function GetPercentile99Data() {
+		if (selectedData === null) {
+			return null;
+		}
 		return selectedData
 			.map(a => ({ x: a.timestamp, y: a.percentile99 * 1000, endpoint: a.endpoint }));
 	}
 
 	function GetCountRequestData() {
+		if (selectedData === null) {
+			return null;
+		}
 		return selectedData
-			.map(a => ({ x: a.timestamp, y: a.percentile95 * 1000, endpoint: a.endpoint }));
+			.map(a => ({ x: a.timestamp, y: a.countrequests, endpoint: a.endpoint }));
 	}
 
 	function GetDbExecTimeData() {
+		if (selectedData === null) {
+			return null;
+		}
 		return selectedData
 			.map(a => ({ x: a.timestamp, y: a.dbExecTime * 1000, endpoint: a.endpoint }));
 	}
 
 	function GetRenderTimeData() {
+		if (selectedData === null) {
+			return null;
+		}
 		return selectedData
 			.map(a => ({ x: a.timestamp, y: a.renderTime * 1000, endpoint: a.endpoint }));
 	}
 
-	function GetMedianData(){
+	function GetMedianData() {
+		if (selectedData === null) {
+			return null;
+		}
 		return selectedData
 			.map(a => ({ x: a.timestamp, y: a.median * 1000, endpoint: a.endpoint }));
 	}
 
+	function GetDownloads() {
+		if (selectedGeneralData === null) {
+			return null;
+		}
+		return selectedGeneralData
+			.map(a => ({ x: a.timestamp, y: a.downloads }));
+	}
+
+	function GetUploads() {
+		if (selectedGeneralData === null) {
+			return null;
+		}
+		return selectedGeneralData
+			.map(a => ({ x: a.timestamp, y: a.uploads }));
+	}
+
+	function GetViews() {
+		if (selectedGeneralData === null) {
+			return null;
+		}
+		return selectedGeneralData
+			.map(a => ({ x: a.timestamp, y: a.views }));
+	}
+
+	function GetUncaughtExceptions() {
+		if (selectedGeneralData === null) {
+			return null;
+		}
+		return selectedGeneralData
+			.map(a => ({ x: a.timestamp, y: a.uncaughtExceptions }));
+	}
+
+	function GetCountTotalRequests() {
+		if (selectedGeneralData === null) {
+			return null;
+		}
+		return selectedGeneralData
+			.map(a => ({ x: a.timestamp, y: a.countTotalRequests }));
+	}
+
+	function GetOutliersSeconds() {
+		if (outliers === null) {
+			return null;
+		}
+		return outliers.map(a => ({ x: a.timestamp, y: a.seconds * 1000, endpoint: a.endpoint, reqinfo: a.reqinfo }));
+	}
+
+	function GetPrivateMemory() {
+		if (selectedGeneralData === null) {
+			return null;
+		}
+		return selectedGeneralData
+			.map(a => ({ x: a.timestamp, y: (a.privateMemory / 1024) / 1024 }));
```

Comment:
Hier is het misschien duidelijker als je een paar constants definieert als `var kb = 1024`, `var mb = kb * 1024`, en die dan gebruikt ipv de literal ` value / 1024 / 1024`